docs/Userguide_login.rst

```diff
@@ -116,45 +317,73 @@ be able to use ``mila`` with ``scp``, ``rsync`` and other such programs.
     evaluated locally before ssh is even executed).
 
 
-Passwordless login
-------------------
-
-To save you some repetitive typing it is highly recommended to set up public
-key authentication, which means you won't have to enter your password every time
-you connect to the cluster.
+.. important::
+    Login nodes are merely *entry points* to the cluster. They give you access
+    to the compute nodes and to the filesystem, but they are not meant to run
+    anything heavy. Do **not** run compute-heavy programs on these nodes,
+    because in doing so you could bring them down, impeding cluster access for
+    everyone.
 
-.. code-block:: bash
+    This means no training scripts or experiments and no compilation of software
+    unless it is small or ends quickly. Do not run anything that demands a
+    sustained large amount of computation or a large amount of memory.
+
+    **Rule of thumb:** Never run a program that takes more than a few seconds on
+    a login node, unless it mostly sleeps or mostly moves data.
+
+    **Examples:** A non-exhaustive list of use-cases, to give a sense of what is
+    and is not allowed on the login nodes:
+
+    - A Python training script is unacceptable on the login nodes.
+      *(Too computationally- and memory-intensive)*
+    - A Python or shell script that downloads a dataset and exits immediately
+      after may be acceptable on the login nodes.
+      *(Mostly moves data)*
+    - A Python hyperparameter search script that uses ``submitit`` to launch
+      jobs and only sleeps waiting for them to end and run other jobs is
+      acceptable on the login nodes.
+      *(Mostly sleeps; The actual jobs run on the compute nodes)*
+    - ``pip install`` of ``vllm`` or ``flash-attn`` from source code on the
+      login nodes is unacceptable (and is likely to fail anyways).
+      *(Takes far too much RAM to compile the CUDA kernels)*
+    - Editing code with ``nano``, ``vim`` or ``emacs`` is acceptable.
+      *(Editors mostly sleep awaiting user keystrokes)*
+    - Copying/moving files with ``cp``, ``mv``, ... is acceptable.
+      *(Mostly moves data)*
+    - Connecting to compute nodes with ``ssh`` is acceptable.
+      *(Mostly sleeps, forwarding keystrokes and ports to/from the node)*
+    - Using ``tmux`` is acceptable.
+      *(Mostly sleeps, managing the processes under its control)*
 
-    # ON YOUR LOCAL MACHINE
-    # You might already have done this in the past, but if you haven't:
-    ssh-keygen  # Press ENTER 3x
+    .. note::
+        In a similar vein, you should not run VSCode remote SSH instances directly
+        on login nodes, because even though they are typically not very
+        computationally expensive, when many people do it, they add up! See
+        :ref:`Visual Studio Code` for specific instructions.
```

Comment:
Does that section mention `mila code`? Couldn't hurt to mention that command here as well IMO.

Reply:
This section already exists in the current documentation. It appears as new because of the number of changes. The reference `:ref:Visual Studio Code` is a link to https://docs.mila.quebec/VSCode.html#visual-studio-code which has to links on how to use `mila code`.